coderd/runtimeconfig/config.go

```diff
@@ -0,0 +1,97 @@
+package runtimeconfig
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	"github.com/spf13/pflag"
+	"golang.org/x/xerrors"
+)
+
+// TODO: comment
+type Value pflag.Value
+
+type Entry[T Value] struct {
+	val T
+	key string
+}
+
+func New[T Value](key, val string) (out Entry[T], err error) {
+	out.Init(key)
+
+	if err = out.Set(val); err != nil {
+		return out, err
+	}
+
+	return out, nil
+}
+
+func MustNew[T Value](key, val string) Entry[T] {
+	out, err := New[T](key, val)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+func (o *Entry[T]) Init(key string) {
+	o.val = create[T]()
+	o.key = key
+}
+
+func (o *Entry[T]) Set(s string) error {
+	if reflect.ValueOf(o.val).IsNil() {
+		return xerrors.Errorf("instance of %T is uninitialized", o.val)
+	}
+	return o.val.Set(s)
+}
+
+func (o *Entry[T]) Type() string {
+	return o.val.Type()
+}
+
+func (o *Entry[T]) String() string {
+	return o.val.String()
+}
+
+func (o *Entry[T]) StartupValue() T {
+	return o.val
+}
+
+func (o *Entry[T]) Resolve(ctx context.Context, r Resolver) (T, error) {
+	return o.resolve(ctx, r)
+}
+
+func (o *Entry[T]) resolve(ctx context.Context, r Resolver) (T, error) {
```

Comment:
Why have this method twice? Why not just `Resolve`?

Reply:
`Coalesce` calls it as well.